Emphasise ways to get help in user emails

mkllnk opened the pull request:
#### What? Why?

Closes #6102

<!-- Explain why this change is needed and the solution you propose.
Provide context for others to understand it. -->

We had a very prominent footer showing how to get in contact with the
local instance people but most users need to get in contact with the
enterprise they are buying from. So removing all those details and
replacing them by a simple "powered by" line will hopefully direct
attention to the shop's contact details.


#### What should we test?
<!-- List which features should be tested and how. -->

Trigger the following emails and confirm that they still look good:


-   spree/order_mailer/cancel_email
-   spree/order_mailer/confirm_email_for_customer
-   spree/user_mailer/confirmation_instructions
-   subscription_mailer/confirmation_summary_email
-   subscription_mailer/placement_summary_email

Remove the email configuration as super admin (/admin/contents/edit) and re-send an order confirmation.
The link should now point to the global website.

#### Release notes
<!-- Write a one liner description of the change to be included in the release notes.
Every PR is worth mentioning, because you did it for a reason. -->

Simplified the order confirmation email amongst some others to direct more attention to shops instead of the Open Food Network instance.
<!-- Please select one for your PR and delete the other. -->
Changelog Category: User facing changes

sauloperez reviewed (approved):
neat :ok_hand:

sauloperez commented:
It looks like we need to include the helper somewhere because it broke another mailre.

Matt-Yorkley commented:
LGTM but as Pau said, there's a couple of build errors that need fixing. Moving to in dev.

mkllnk commented:
Ah, thanks. Good that I rebased then! I fixed that spec and will move this to test-ready if the whole build passes.

filipefurtad0 commented:
Hey @mkllnk ,

Thank you for addressing this.

I checked the emails you mentioned, looking specifically whether the footer is changed and everything else looked as before:

**1)** spree/order_mailer/cancel_email - OK

![image](https://user-images.githubusercontent.com/49817236/108411540-a6ecf880-7220-11eb-8c3a-7f74be187c30.png)

**2)** spree/order_mailer/confirm_email_for_customer - OK

![image](https://user-images.githubusercontent.com/49817236/108415000-cab23d80-7224-11eb-9f03-029916958bc0.png)

**3)** spree/user_mailer/confirmation_instructions  - this one is breaking, after this PR:

https://app.bugsnag.com/yaycode/openfoodnetwork-uk/errors/602ec51df8503d0018306cc1?event_id=602ec51d0069ae371de30000&i=sk&m=nw

I tested this before, and after this PR, in different stagings - it really looks like it is introduced in this code.

**4)** subscription_mailer/confirmation_summary_email

![image](https://user-images.githubusercontent.com/49817236/108413075-78701d00-7222-11eb-9ba8-65d91f7942fd.png)

**5)** subscription_mailer/placement_summary_email

![image](https://user-images.githubusercontent.com/49817236/108412753-16171c80-7222-11eb-9d72-fee9141cdbc5.png)

**6)** Removing the email configuration as super admin (/admin/contents/edit): the footer on order confirmation emails now point to the global website (instead of the superadmin mail address) - all good here.


This is causing the sign-up confirmation email to fail Maikel (**testcase 3**); I think this is quite relevant email, so I'm moving this back to In Dev, ok?

mkllnk commented:
> This is causing the sign-up confirmation email to fail

Great catch, @filipefurtad0! :fishing_pole_and_fish: 

I added a unit test for this so that it won't happen again. I moved this back to review but I think that one approval is enough this time. Only the last commit is new.

sauloperez reviewed (approved):
:+1: I'll merge once we get past that flaky spec.

mkllnk commented:
Shouldn't we test this again?

filipefurtad0 commented:
Thanks for adding that spec @mkllnk ,

I made a quick test anyway and the email footer looks good:

![image](https://user-images.githubusercontent.com/49817236/109319346-ac60c900-7846-11eb-9360-671fe9eb9bae.png)

The translation is missing but the key was added to `en.yml` - we're good to go! :tada:
